packages/terra-folder-tree/src/subcomponents/FolderTreeItem.jsx

```diff
@@ -0,0 +1,122 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import { injectIntl } from 'react-intl';
+
+import Spacer from 'terra-spacer';
+import Arrange from 'terra-arrange';
+import { IconFolder } from 'terra-icon';
+import ThemeContext from 'terra-theme-context';
+
+import styles from './FolderTreeItem.module.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The label for the folder tree item.
+   */
+  label: PropTypes.string.isRequired,
+  /**
+   * The icon to display to the left of the name.
+   */
+  icon: PropTypes.element,
+  /**
+   * List of FolderTree.Items to display in a subfolder when this FolderTreeItem is clicked.
+   */
+  subfolderItems: PropTypes.arrayOf(PropTypes.element),
+  /**
+   * Whether or not the item is selected.
```

Comment:
```suggestion
   * Whether or not the item is selected. Since this component has the appearance of a radio button group, only one item should be selected at a time.
```